cmd/agent/api/agent/agent.go

```diff
@@ -179,6 +191,64 @@ func getStatus(w http.ResponseWriter, r *http.Request) {
 	w.Write(jsonStats)
 }
 
+func streamLogs(w http.ResponseWriter, r *http.Request) {
+	log.Info("Got a request for stream logs.")
+	w.Header().Set("Transfer-Encoding", "chunked")
+	logMessageReceiver := logs.GetMessageReceiver()
+	if logMessageReceiver == nil {
+		http.Error(w, "The logs agent is not running", 405)
+		w.(http.Flusher).Flush()
+		log.Info("Logs agent is not running - can't stream logs")
+		return
+	}
+
+	if !logMessageReceiver.SetEnabled(true) {
+		http.Error(w, "Another client is already streaming logs.", 405)
+		w.(http.Flusher).Flush()
+		log.Info("Logs are already streaming. Dropping connection.")
+		return
+	}
+	defer logMessageReceiver.SetEnabled(false)
+
+	var filters diagnostic.Filters
+
+	if r.Body != http.NoBody {
+		body, err := ioutil.ReadAll(r.Body)
+		if err != nil {
+			http.Error(w, log.Errorf("Error while reading HTTP request body: %s", err).Error(), 500)
```

Comment:
Nit. It looks like you are using `log.Errorf` to generate an error string, any reason not to use `fmt.Sprintf` instead for clarity? Understood about side effect of logging the error but it might be confusing to read for some.

Reply:
I copied this code from line:99 since they are both handling json the same way (tried to make a helper function it but it didn't really work due to Unmarshal expecting a different type). I can totally change it if your suggestion is preferable :)

Reply:
Ah, gotcha np, makes sense, if it follows the existing style 👍